### ISSUE TITLE
Remove `sentry-cli` from the repository

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-fastlane-plugin#165]
In this PR we've added more archictecutre/platform supports for sentry-cli:
* https://github.com/getsentry/sentry-fastlane-plugin/pull/164/files

This will eventually bloat the repository significantly with the updates. 

## Proposed solution

Download sentry-cli on the first build. Check the checksum of the downloaded version against the checksum stored in the repo.

**Comment by denrase** [comment]
@bruno-garcia I could take over this issue if you don't have anyone else in mind already.

**Comment by philipphofmann** [comment]
Please take it @denrase.

**Comment by philipphofmann** [comment]
Maybe we could apply a similar approach as we do in https://github.com/getsentry/sentry-dart-plugin/pull/49.

**Comment by marandaneto** [comment]
We've agreed that downloading on the fly is actually an improvement since you don't bloat the repo and the user has to download only the needed binary, although, if it's working and it's not a source of complaints, we can prioritize more important issues, and this is eventually done when there are no more p1 for example.
cc @denrase @kahest @philipphofmann

**Comment by philipphofmann** [comment]
I agree with lowering the priority for this issue.

**Comment by bruno-garcia** [comment]
The idea is to download when packaging. Downloading on the uses machine has the downside of slowing down the first build,. Requiring internet access on the build time as opposed to "restore deps" phase and brings security concerns (unless we pin ssl cert, which isn't good to maintain).

So the trade off is to check in the Sha in the repo and in CI when packaging, download the binary and verify the checksum. The final artifact will bundle it.

**Comment by philipphofmann** [comment]
We could consider using https://git-lfs.com/.

**Comment by philprime** [comment]
Is there any reason for us to still consider this? It was originally requested about 3 years ago, so if it wasn't so important for so long we might just not do it.

**Comment by philippsuess** [comment]
Yes, I'd like to close it for now. We can open a new issue if it comes up again.
